cli/run_util.go

```diff
@@ -540,14 +583,66 @@ func (r WorkflowRunner) activateAndRunSteps(
 	secrets []envmanModels.EnvironmentItemModel,
 	isLastWorkflow bool,
 	tracker analytics.Tracker,
-	workflowIDProperties coreanalytics.Properties) models.BuildRunResultsModel {
+	workflowIDProperties coreanalytics.Properties,
+	workflowID string,
+) models.BuildRunResultsModel {
 	log.Debug("[BITRISE_CLI] - Activating and running steps")
 
 	if len(workflow.Steps) == 0 {
 		log.Warnf("%s workflow has no steps to run, moving on to the next workflow...", workflow.Title)
 		return buildRunResults
 	}
 
+	err := tools.EnvmanInit(configs.InputEnvstorePath, true)
+	if err != nil {
+		log.Debugf("Couldn't initialize envman.")
+	}
+	err = tools.EnvmanAddEnvs(configs.InputEnvstorePath, *environments)
+	if err != nil {
+		log.Debugf("Couldn't add envs.")
+	}
+	envList, err := tools.EnvmanReadEnvList(configs.InputEnvstorePath)
+	if err != nil {
+		log.Debugf("Couldn't read envs from envman.")
+	}
+
+	serviceContainers, err := r.dockerManager.StartServiceContainers(workflow.Services, workflowID, envList)
```

Comment:
I'd be more comfortable with only executing all of this new code if and only if the container block is defined and valid 😅

Reply:
This particular part is for the service containers and it will only execute if they are defined, if the workflow.Services is empty it will be a NOOP operation.

Reply:
But I'm not so sure about the rest of the envman stuff above. It's pretty legacy stuff, I'm not sure if `envman init` is idempotent or not, this way it's going to get initialized twice even if the Docker feature is not used.